homeassistant/components/rest/__init__.py

```diff
@@ -167,6 +169,10 @@ def create_rest_data_from_config(hass, config):
         resource_template.hass = hass
         resource = resource_template.async_render(parse_result=False)
 
+    if payload_template is not None:
+        payload_template.hass = hass
+        payload = payload_template.async_render(parse_result=False)
```

Comment:
Shouldn't this be re-rendered each time the sensor's state is refreshed, like is done for the `headers` and `params` templates?